dataland-frontend/tests/e2e/specs/search-company-framework-data/SearchCompaniesForFrameworkDataDropdownFilter.ts

```diff
@@ -103,36 +104,49 @@ describe("As a user, I expect the search functionality on the /companies page to
     "Checks that the sector filter synchronises between the search bar and the drop down and works",
     { scrollBehavior: false },
     () => {
-      const demoCompanyToTestFor = companiesWithEuTaxonomyDataForNonFinancials[0].companyInformation;
+      const demoCompanyToTestFor = assertDefined(
+        companiesWithEuTaxonomyDataForNonFinancials.find((it) => it.companyInformation.sector !== undefined)
+      ).companyInformation;
       const demoCompanyWithDifferentSector = companiesWithEuTaxonomyDataForNonFinancials.find(
-        (it) => it.companyInformation.sector !== demoCompanyToTestFor.sector
+        (it) =>
+          it.companyInformation.sector !== demoCompanyToTestFor.sector && it.companyInformation.sector !== undefined
       )!.companyInformation;
 
+      expect(demoCompanyToTestFor.sector).to.not.be.undefined;
       cy.ensureLoggedIn();
       cy.intercept("**/api/companies/meta-information").as("companies-meta-information");
-      cy.visit(`/companies?input=${demoCompanyToTestFor.companyName}&sector=${demoCompanyWithDifferentSector.sector}`)
+      cy.visit(`/companies?input=${demoCompanyToTestFor.companyName}&sector=${demoCompanyWithDifferentSector.sector!}`)
         .wait("@companies-meta-information")
         .get("div[class='col-12 text-left']")
         .should("contain.text", failureMessageOnAvailableDatasetsPage)
         .get("#sector-filter")
         .click()
         .get('input[placeholder="Search sectors"]')
-        .type(`${demoCompanyToTestFor.sector}`)
+        .type(`${demoCompanyToTestFor.sector!}`)
         .get("li")
-        .contains(RegExp(`^${demoCompanyToTestFor.sector}$`))
+        .contains(RegExp(`^${demoCompanyToTestFor.sector!}$`))
```

Comment:
Ist die Regex nicht overkill?  `contains` macht doch von sich aus ein vernünftiges substring-matching

Reply:
Weißt du was ^ bewirkt? Ich glaube $ sorg dafür, dass sich der String am Ende befindet. Von daher ist es schonmal mehr als ein einfache string based contains.

Reply:
Ok habe nochmal nachgschaut.  Durch die RegEx hier wird ein exaktes matching gemacht. kein substring matching.
Ist also "strenger".   Kann von mir aus bleiben um den Fall auszuschließen, dass der companyName eines fakeFixtures zufällig der substring eines anderen companyNames ist.

Reply:
Aaaah, right. Contains delivers a DOM element what `.should("have.text"), of course does not. So this will stay as is.